etc/lmod-setup.sh

```diff
@@ -35,6 +35,10 @@ elif [ "$L_MACHINE" = odin ]; then
    module --initial_load --no_redirect restore
    export MODULEPATH="/oldscratch/ywang/external/hpc-stack/modulefiles/mpi/intel/2020/cray-mpich/7.7.16:/oldscratch/ywang/external/hpc-stack/modulefiles/compiler/intel/2020:/oldscratch/ywang/external/hpc-stack/modulefiles/core:/oldscratch/ywang/external/hpc-stack/modulefiles/stack:/opt/cray/pe/perftools/21.02.0/modulefiles:/opt/cray/ari/modulefiles:/opt/cray/pe/craype-targets/default/modulefiles:/opt/cray/pe/modulefiles:/opt/cray/modulefiles:/opt/modulefiles"
 
+elif [ "$L_MACHINE" = wcoss2 ]; then
+   module purge
+   module load envvar/1.0
```

Comment:
Can this go to the modulefiles instead? Unless it is loading Lmod and it is something that can be loaded in modulefiles, I think it should go there.

Reply:
@danielabdi-noaa, when a user logs in wcoss2, the module `envvar/1.01` is automatically loaded by default. In the wcoss2 user wiki, the following is described in the section of "Lmod and User Environments": `Please note that when doing a module purge you need to reload envvar, module load envvar/1.0, in your environment (interactive or batch) to get the appropriate default modules. NCO has implemented the standard of doing a “module reset”. This implements the module purge and module load of envvar/1.0 in one step.` So I think this command would be better to be placed here. What do you think about this?

Reply:
I think `module load envar/1.0` can go to the top of `build_wcoss2_*`. Have you tested this change and found it to cause a problem? It looks like it is loading default modules which should not be any different from loading custom modulefiles of SRW. It is not changing the module app, as is the case for example on `gaea/odin` going from Cray->Lmod or on macos where you have no Lmod when you start. Also, the way you have it now if the user does a regular `module purge` on the command line, he has to follow it up with `module load envvar/1.0`, which would be avoided if you have the latter in `build_wcoss2`. But for this to be the case, you have to test  loading envvar in `build_wcoss2` works in the first place ...

Reply:
Once again, machine specific things need to be isolated to the machine-specific configuration.  I agree with @danielabdi-noaa. This belongs in the wcoss configuration module file.  It doesn't make sense here, and pollutes the code with machine-specific things in places users and developers don't expect.

Reply:
@danielabdi-noaa, okay, I'll test it and move it to `build_wcoss2_*` unless it cases any issues.

Reply:
@danielabdi-noaa , @christopherwharrop-noaa , what do you think of replacing "module purge and module load envvar" with "module reset"?

Reply:
If you are putting it in the `build_wcoss*` modulefile, then you can do whatever you want there as that is completely isolated behavior for that particular machine configuration.  My objection is the test for machine name in a non-machine-specific config/module file.

Reply:
Do you mean you don't like `module reset`, either? It is an official nco standard on wcoss2.

Reply:
There are really two problems.  First, manipulation of modules really belongs in the machine-specific modulefiles.  The other big problem is the `elif [ "$L_MACHINE" = wcoss2 ]`.  That is bad design and should be fixed.  The conditional tests in this file for which machine you are on needs to be removed and moved from the code/scripts to machine-specific configurations.  But, it appears that you are just following the existing template here, and so I can approve this as is for now because fixing this design problem is beyond the scope of your proposed changes.

Reply:
@christopherwharrop-noaa, now I got what your point was. You pointed out that the if-statement itself in the 'lmod-setup.sh' file would cause the issues you mentioned. Is this correct? I thought that you were talking about the command 'module load envvar'. I'll remove 'wcoss2' from the if-statement. Sorry for my late grasp! :) :)

Reply:
@chan-hoo I think it is Ok both ways. It is desirable if sourcing `etc/lmod-setup.sh/csh` can be avoided, one less step. On most systems that is actually the case and all you get if you do source it is the `module purge` in the `else:` condition.  In my opinion, those two etc files should be restricted to setting up Lmod, and loading default modules (envvar) can be considered part of setting up Lmod. If it was upto me, I would rather have the user load Lmod manually before building SRW :), but I know your stand on this. Thanks for addressing the requests.